web/pages/charity/index.tsx

```diff
@@ -3,26 +3,41 @@ import { useState, useMemo } from 'react'
 import { charities as charityList } from '../../../common/charity'
 import { CharityCard } from '../../components/charity/charity-card'
 import { Col } from '../../components/layout/col'
+import { Spacer } from '../../components/layout/spacer'
 import { Page } from '../../components/page'
 import { Title } from '../../components/title'
-
-const charities = charityList.map((charity) => ({
-  ...charity,
-  raised: 4001,
-}))
+import { useAllCharityTxns } from '../../hooks/use-charity-txns'
 
 export default function Charity() {
+  const allCharityTxn = useAllCharityTxns()
+  const charities = useMemo(() => {
+    return charityList.map((charity) => {
+      const txns = allCharityTxn.filter((txn) => txn.toId === charity.id)
+      return {
+        ...charity,
+        raised: _.sumBy(txns, (txn) => txn.amount),
```

Comment:
I'd prefer to store raised in a separate map (from charity id to amount raised) instead of creating modified charity objects. But this works so fine to keep.